lib/client.js

```diff
@@ -192,6 +254,172 @@ Client.prototype = {
+    var deferred = Promise.defer();
+    if (this._context) {
+      deferred.resolve(this._context);
+    } else {
+      this.on('app.registered', function() {
+        deferred.resolve(this._context);
+      }.bind(this));
+    }
+    return deferred.promise;
+  },
+
+  /// #### client.get(paths)
+  ///
+  /// Get data from the UI asynchronously.
+  ///
+  /// ##### Arguments
+  ///
+  ///   * `paths` an array of path strings or a single path string
+  ///
+  /// ##### Returns
+  ///
+  /// A [Promises/A+](https://promisesaplus.com) conformant `promise` object.
+  ///
+  /// ```javascript
+  /// var client = ZAFClient.init();
+  ///
+  /// client.ui(['ticket.subject', 'ticket.fields.custom_field_1234.value').then(function(data) {
+  ///   console.log(data); // ['Help, my printer is on fire', 'Printer on Fire']
+  /// });
+  /// ```
+  get: function(paths) {
+    var wasArray = Array.isArray(paths);
+    if (!wasArray) {
+      paths = [ paths ];
+    }
+    var promise = this.call('data.get', paths);
+    return (wasArray) ? promise : promise.then(function(data) {
+      return paths[0].split('.').reduce(function(memo, pathSegment) {
+        return memo[pathSegment];
+      }, data);
+    });
+  },
+
+  /// #### client.set(key, val)
+  /// #### client.set(obj)
+  ///
+  /// Set data in the UI asynchronously.
+  ///
+  /// ##### Arguments
+  ///
+  ///   * `key` the path to which to set the value `val`
+  ///   * `obj` a json object containing the keys and values to update
+  ///
+  /// ##### Returns
+  ///
+  /// A [Promises/A+](https://promisesaplus.com) conformant `promise` object returning the updated data.
+  ///
+  /// ```javascript
+  /// var client = ZAFClient.init();
+  ///
+  /// client.set('ticket', { subject: 'Printer Overheating Incident', type: 'incident' }).then(function(ticket) {
+  ///   console.log(ticket);
+  /// });
+  /// ```
+  set: function(key, val) {
+    var obj = key;
+    if ('string' === typeof key) {
+      // namespace the setted object
+      obj = key.split('.').reverse().reduce(function(memo, pathSegment) {
+        var ret = {};
+        ret[pathSegment] = memo;
+        return ret;
+      }, val);
+
+      return this.call('data.set', obj).then(function(ret) {
+        // denamespace the return value
+        return key.split('.').reduce(function(memo, pathSegment) {
+          return memo[pathSegment];
+        }, val);
+      });
+    } else {
+      return this.call('data.set', obj);
+    }
+  },
+
+  /// #### client.call(name, [data])
+  ///
+  /// Remotely call an API from the host framework.
+  ///
+  /// ##### Returns
+  ///
+  /// A [Promises/A+](https://promisesaplus.com) conformant `promise` object.
+  ///
+  /// ```javascript
+  /// var client = ZAFClient.init();
+  ///
+  /// client.call('modal.show', { url: 'https://www.example.com/modal.html' }).then(function(data) {
+  ///   console.log(data); // { width: 200, height: 200 }
+  /// });
+  /// ```
+  call: function(name) {
```

Comment:
got some suggestions? call does fit the best in my mind at least, but I do see how it could be confused with Function.prototype.call